src/cpp/fastdds/xtypes/dynamic_types/idl_parser/IdlParser.hpp

```diff
@@ -0,0 +1,1725 @@
+            if (state["union_labels"].empty() || state["union_labels"].back() == ';')
+            {
+                state["union_labels"] += "default";
+            }
+            else
+            {
+                state["union_labels"] += ",default";
+            }
+        }
+        else
+        {
+            if (state["union_labels"].empty() || state["union_labels"].back() == ';')
+            {
+                state["union_labels"] += label;
+            }
+            else
+            {
+                state["union_labels"] += "," + label;
+            }
+        }
+
+        if (state["union_discriminant"].empty())
+        {
+            state["union_discriminant"] = state["type"];
+        }
+    }
+
+};
+
+template<>
+struct action<switch_case>
+{
+    template<typename Input>
+    static void apply(
+            const Input& in,
+            Context* ctx,
+            std::map<std::string, std::string>& state,
+            std::vector<traits<DynamicData>::ref_type>& operands)
+    {
+        Module& module = ctx->module();
+
+        state["union_labels"] += ";";
+        state["union_member_types"] += ";";
+        state["union_member_names"] += ";";
+    }
+
+};
+
+template<>
+struct action<union_def>
+{
+    template<typename Input>
+    static void apply(
+            const Input& in,
+            Context* ctx,
+            std::map<std::string, std::string>& state,
+            std::vector<traits<DynamicData>::ref_type>& operands)
+    {
+        Module& module = ctx->module();
+        const std::string& union_name = state["union_name"];
+
+        DynamicTypeBuilderFactory::_ref_type factory {DynamicTypeBuilderFactory::get_instance()};
+        TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+        DynamicType::_ref_type discriminant_type = ctx->get_type(state, state["union_discriminant"]);
+        type_descriptor->kind(TK_UNION);
+        type_descriptor->name(union_name);
+        type_descriptor->discriminator_type(discriminant_type);
+        DynamicTypeBuilder::_ref_type builder {factory->create_type(type_descriptor)};
+
+        std::vector<std::string> label_groups = ctx->split_string(state["union_labels"], ';');
+        std::vector<std::string> types = ctx->split_string(state["union_member_types"], ';');
+        std::vector<std::string> names = ctx->split_string(state["union_member_names"], ';');
+
+        std::vector<std::vector<int32_t>> labels;
+        int default_label_index = 0;
+        for (size_t i = 0; i < label_groups.size(); i++)
+        {
+            if (label_groups[i].empty()) continue; // Skip empty strings
+            std::vector<std::string> numbers_str = ctx->split_string(label_groups[i], ',');
+            std::vector<int32_t> numbers;
+            for (const auto& num_str : numbers_str)
+            {
+                if (num_str == "default")
+                {
+                    numbers.push_back(std::numeric_limits<int32_t>::max());
```

Comment:
With the `default` label it is not needed to add a `label` with value `std::numeric_limits<int32_t>::max()`. For this case the `DynamicTypeMember` only need to have `is_default_label` set to `true`.

Reply:
Done.